### PR TITLE
Adding a delete button to the Trip List Page

### LINKED ISSUE #77: Add delete trip button to TripListPage.
For each trip, like each trip card in the list should have a minus button or whatever.

### PR DESCRIPTION
**Pull request by anfee1**
Still needs testing.

Closes: #77

### CODE REVIEW

**Comment by anfee1** [comment]
I think we need to add a delete method to the TripService as well unless we were supposed to be doing something else.

File: frontend/ui/lib/widgets/trip_list_widget.dart [hunk 1]
```diff
@@ -35,8 +36,33 @@ class TripListWidget extends StatelessWidget {
...
+                  )),
             ))
         .toList();
     return ListView(children: listItems);
   }
+
+  _showDialog(BuildContext context) {
+    VoidCallback continueCallBack = () => {
+          Navigator.of(context).pop(),
+          //I don't think we have a delete trip method.
```
Comment (bsidhom) [inline frontend/ui/lib/widgets/trip_list_widget.dart:55]: Looks like we need to add one on the Dart side. We _do_ have a delete-trip servlet implementation (`TripServlet.doDelete`).

File: frontend/ui/lib/widgets/trip_list_widget.dart [hunk 2]
```diff
@@ -35,8 +36,33 @@ class TripListWidget extends StatelessWidget {
                 Navigator.pushNamed(
                     context, Router.createTripViewRoute(trip.id));
               },
+              trailing: Tooltip(
+                  message: 'Click here to delete trip.',
```
Comment (bsidhom) [inline frontend/ui/lib/widgets/trip_list_widget.dart]: Make the `IconButton` the top-level widget here and make the tooltip a property of that button.

**Review by bsidhom — DISMISSED** [review]
LGTM

**Review by bsidhom — APPROVED** [review]
LGTM

**Review by cli1903 — APPROVED** [review]

**Pull request merged**